### PR TITLE
Ear 307 skip logic front end

### PR DESCRIPTION
**Pull request by martyncolmer**
Well.
Theres a lot to review here.

Spin the app up and ensure the functionality behaves as expected.

Review the Skip logic components and feedback any obvious coding issues or suggestions.

### CODE REVIEW

File: eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/SkipLogicEditor/SkipConditionEditor/index.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  small: true,
+})`
+  margin-left: auto;
+  padding: 0.2em;
+`;
+
+export class UnwrappedSkipConditionEditor extends React.Component {
+  static propTypes = {
+    pageId: PropTypes.string,
+    expressionGroup: PropTypes.object, // eslint-disable-line
```
Comment (MattElcock) [inline eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/SkipLogicEditor/SkipConditionEditor/index.js:73]: We should probably look into disabling this eslint rule as we use `PropTypes.object` quite a bit. I think we can add a config file specifying which rules we want.

File: eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/index.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { shallow } from "enzyme";
+
+import { UnwrappedSkipLogicPage as SkipLogicPage } from "./";
+
+import NoSkipConditions from "./NoSkipConditions";
+import SkipLogicEditor from "./SkipLogicEditor";
+
+describe("Skip Condition Page", () => {
+  it("should show the no skip sondition message when there is no skip sonditions for a page", () => {
```
Comment (MattElcock) [inline eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/index.test.js:10]: Spelling

```suggestion
  it("should show the no skip condition message when there is no skip conditions for a page", () => {
```

File: eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/index.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const wrapper = shallow(
+      <SkipLogicPage
+        page={{ id: "1", displayName: "test", skipConditions: null }}
+        createSkipCondition={jest.fn()}
+      />
+    );
+    expect(wrapper.find(NoSkipConditions).exists()).toBe(true);
+  });
+
+  it("should call create skip sondition with the page id when add skip sondition button is clicked", () => {
```
Comment (MattElcock) [inline eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/index.test.js:20]: Spelling

```suggestion
  it("should call create skip condition with the page id when add skip condition button is clicked", () => {
```

File: eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/SkipLogicEditor/SkipConditionEditor/index.js (hunk 1 above)
Comment (martyncolmer) [reply eq-author/src/App/page/Logic/SkipLogic/SkipLogicPage/SkipLogicEditor/SkipConditionEditor/index.js:73]: Agreed

**Comment by martyncolmer** [comment]
> On a checkbox, if I make a condition which selects any or all of a subset of answers, then change the condition to be 'unanswered', the right side remains when it should disapere.
> 
> ```
> "skipConditions": [
>             {
>               "id": "aeb511f6-712b-4e99-a5af-1c883b5e7728",
>               "expressions": [
>                 {
>                   "id": "c8f29621-460f-46e8-b2e5-a804c1c9d74c",
>                   "condition": "Unanswered",
>                   "left": {
>                     "type": "Answer",
>                     "answerId": "559d63a3-6f20-4a1f-a9d3-4ac486c2813f"
>                   },
>                   "right": {
>                     "type": "SelectedOptions",
>                     "optionIds": [
>                       "da9173e2-a9f7-40b1-99ff-2bbe44e93654"
>                     ]
>                   }
>                 }
>               ]
>             }
>           ]
> ```

This is the same behave for routing. When we publish the right condition is ignored.

**Pull request merged**